apps/src/music/SoundsPanel.jsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from './soundsPanel.module.scss';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+
+const getSoundRowClassName = (currentValue, folderPath, soundSrc) => {
+  if (currentValue === folderPath + '/' + soundSrc) {
+    return styles.soundRowSelected;
+  } else {
+    return styles.soundRow;
+  }
+};
+
+const getPreviewClassName = (playingPreview, folderPath, soundSrc) => {
+  if (playingPreview === folderPath + '/' + soundSrc) {
+    return styles.previewPlaying;
+  } else {
+    return styles.preview;
+  }
+};
+
+const getIcon = type => {
+  const typeToIcon = {
+    beat: 'volume-up',
+    bass: 'headphones',
+    lead: 'music',
+    fx: 'asterisk'
+  };
+  return typeToIcon[type];
+};
+
+const getIconClassName = type => {
+  return styles['icon-' + type];
+};
+
+const SoundsPanel = ({
+  library,
+  currentValue,
+  playingPreview,
+  onSelect,
+  onPreview
+}) => {
+  const group = library.groups[0];
+
+  return (
+    <div className={styles.soundsPanel}>
+      {group.folders.map((folder, folderIndex) => {
+        return (
+          <div
+            className={folderIndex === 0 ? styles.folderFirst : styles.folder}
+            key={folderIndex}
+          >
+            <div className={styles.folderName}>{folder.name}</div>
+            {folder.sounds.map((sound, soundIndex) => {
+              return (
+                <div
+                  className={getSoundRowClassName(
+                    currentValue,
+                    folder.path,
+                    sound.src
+                  )}
+                  key={soundIndex}
+                  onClick={() => onSelect(folder.path + '/' + sound.src)}
+                >
+                  <div className={styles.soundRowLeft}>
+                    <FontAwesome
+                      icon={getIcon(sound.type)}
+                      className={getIconClassName(sound.type)}
+                    />
+                  </div>
+                  <div className={styles.soundRowMiddle}>{sound.name}</div>
+                  <div className={styles.soundRowRight}>
+                    <FontAwesome
+                      icon={'play-circle'}
+                      className={getPreviewClassName(
+                        playingPreview,
```

Comment:
Rather than always showing a play button here, I expected the play button to change to a stop button (`'circle-stop'` I think). Currently, if you click the icon while a preview is playing, it select that sound's value for the block field. Considering that we support jumping between different sound previews, it seems like we might also want to support stopping a sound preview on demand.

Reply:
This is a good point, and agreed that toggling to a stop button is likely a good design.  I avoided that for this first pass just to keep things a bit simpler.  Am definitely open to adding it.

Great catch that clicking a preview button while it's dimmed, because we're already playing it, just selects the sound.  I've fixed that.